pkg/planner/core/operator/logicalop/logical_join.go

```diff
@@ -1284,14 +1284,27 @@ func (p *LogicalJoin) extractOnCondition(conditions []expression.Expression, der
 	return p.ExtractOnCondition(conditions, p.Children()[0].Schema(), p.Children()[1].Schema(), deriveLeft, deriveRight)
 }
 
+func extractTableAlias(cteList []*ast.CommonTableExpression, p base.Plan, parentOffset int) *utilhint.HintedTable {
+	alias := util.ExtractTableAlias(p, parentOffset)
+	if alias != nil {
+		for _, cte := range cteList {
+			if cte.Name.L == alias.TblName.L {
+				// with CTE, this offset cannot be smaller than 1.
+				alias.SelectOffset = max(1, alias.SelectOffset-1)
```

Comment:
why we need alias.SelectOffset minus 1 here？

Reply:
```with cte as (select a, count(b) from t2 group by a) select /*+ INL_JOIN(cte) */  * from t1, cte where t1.a=cte.a;```

hint will be Query block Offset -> 1. and CTE will get query block offset -> 2 because it is in join. but in fact, CTE table is a CTE, it can get small query block offset.

Reply:
In fact, the challenge here lies in determining how to handle multiple CTEs and deeper joins, which requires further investigation using fuzz testing to uncover more issues.